File: 0101_0200/0140/0140.py
from math import inf
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.readline())
    m = [[int(x) if x != "100000" else inf for x in f.readline().split()] for i in range(n)]
    for k in range(n):
        for i in range(n):
            for j in range(n):
                if m[i][k] < inf and m[k][j] < inf and m[i][k] + m[k][j] < m[i][j]:
                    m[i][j] = m[i][k] + m[k][j]
    q.write("YES" if any(m[i][i] < 0 for i in range(n)) else "NO")

File: 0101_0200/0140/0140.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, *m;
	char *s = "NO";

	fscanf(f, "%d", &n);
	m = (int*)malloc(sizeof(int) * n * n);
	for (int i = 0; i < n * n; ++i)
	{
		fscanf(f, "%d", m + i);
		if (m[i] == 100000)
			m[i] = INT_MAX;
	}
	for (int k = 0; k < n; ++k)
		for (int i = 0; i < n; ++i)
			for (int j = 0; j < n; ++j)
				if (m[i * n + k] < INT_MAX && m[k * n + j] < INT_MAX && m[i * n + k] + m[k * n + j] < m[i * n + j])
					m[i * n + j] = m[i * n + k] + m[k * n + j];
	for (int i = 0; i < n; ++i)
		if (m[i * n + i] < 0 && (s = "YES"))
			break;
	fprintf(q, s);
	return 0;
}
